controller/generic/metacontroller.go

```diff
@@ -290,38 +322,56 @@ func (mc *ConfigMetaController) wait(condition func() (bool, error)) error {
 // binary
 //
 // NOTE:
-//	This method is used as a condition and hence can be invoked
-// more than once.
+//	This method is used as a condition and is repeatedly executed
+// under a loop till this condition is not met.
 func (mc *ConfigMetaController) startAllWatchControllers() (bool, error) {
-	// In this metacontroller, we are only responsible for
-	// starting/stopping the relevant watch based controllers
+	var errs []string
+	// This logic is responsible to start all the generic controllers
+	// configured in the config file
 	for _, conf := range mc.Configs {
 		key := conf.AsNamespaceNameKey()
 		if _, ok := mc.WatchControllers[key]; ok {
 			// Already added; perhaps during earlier condition
 			// checks
 			continue
 		}
-		// watch controller i.e. a controller based on the resource
-		// specified in the **watch field** of GenericController
+		// Initialize the GenericController
 		wc, err := NewWatchController(
 			mc.ResourceManager,
 			mc.DynClientset,
 			mc.DynInformerFactory,
 			conf,
 		)
 		if err != nil {
-			return false, errors.Wrapf(
-				err,
-				"Failed to init %s: %s",
-				key,
-				mc,
+			errs = append(
```

Comment:
are we able to identify only controller which have failed and restart only those ?

Reply:
Yes. Ths retry logic tries to start the failed controllers only. It skips the controllers that are already running.